hail/src/main/scala/is/hail/expr/ir/InferPType.scala

```diff
@@ -309,6 +350,27 @@ object InferPType {
           InferPType(v._2, env)
           v._2.pType2
       }):_*)
+      case MakeArray(irs, _) => {
+        val it = irs.iterator
+        val head = it.next()
+        InferPType(head, env)
+
+        var allRequired = head.pType2.required
```

Comment:
two problems -- first, `irs` could be empty, so this needs to handle that case by returning something (deepInnerRequired(true)?)

second, prefer functional code (the allRequired bit isn't used, too):
```scala
val elementTypes = irs.map { elt => 
  InferPType(elt, env)
  elt.pType2
}

PArray(getNestedElementPtypes(elementTypes), true)
```

Reply:
If empty, what is the element type, PType(required)?

Reply:
So in the case that I'm trying to unify something that effectively doesn't have an elementType, with something that does, it seems easiest to consider the empty body to be NA( ) of the elementType of the non-missing things.